### ISSUE TITLE
Trying to plot the example code causes an error

### ISSUE THREAD

**Issue by samaloney** [EnsembleGovServices/kamodo-core#54]
hey I'm one of the JOSS review and just going through the review actions now. I was trying to work through the example in the repo readme but running into an error?

## Describe the bug
Following the initial example from repo readme causes an error at the plot step 

## To Reproduce

```python
from kamodo import kamodofy
import numpy as np

x = np.linspace(-np.pi, np.pi, 25)
y = np.linspace(-np.pi, np.pi, 30)
xx, yy = np.meshgrid(x,y)
points = np.array(zip(xx.ravel(), yy.ravel()))

@kamodofy(units = 'km/s')
def fvec(rvec = points):
    ux = np.sin(rvec[:,0])
    uy = np.cos(rvec[:,1])
    return np.vstack((ux,uy)).T

from kamodo import Kamodo

kamodo = Kamodo(fvec = fvec)
kamodo

kamodo.fvec(np.array([[-1,1]]))

kamodo['gvec[m/s]'] = 'fvec'

kamodo.gvec(np.array([[-1,1]]))

import plotly.io as pio

fig = kamodo.plot('fvec')
```

Error output
```
In [7]: fig = kamodo.plot('fvec')
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [7], in <cell line: 1>()
----> 1 fig = kamodo.plot('fvec')

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/kamodo/kamodo.py:1044, in Kamodo.plot(self, plot_partial, *variables, **figures)
   1042 if len(figures) == 1:
   1043     variable, kwargs = list(figures.items())[0]
-> 1044     fig = self.figure(variable, **kwargs)
   1045     # if fig['chart_type'] is None:
   1046     #     raise AttributeError("No chart_type for this trace")
   1047     # else:
   1048     #     if self.verbose:
   1049     #         print('chart type:', fig['chart_type'])
   1050     return go.Figure(
   1051         data=fig['data'],
   1052         layout=fig['layout'])

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/kamodo/kamodo.py:966, in Kamodo.figure(self, variable, indexing, **kwargs)
    964 def figure(self, variable, indexing='ij', **kwargs):
    965     """Generates a plotly figure for a given variable and keyword arguments"""
--> 966     result = self.evaluate(variable, **kwargs)
    967     signature = self.signatures[variable]
    968     units = signature['units']

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/kamodo/kamodo.py:923, in Kamodo.evaluate(self, variable, *args, **kwargs)
    921     if self.verbose:
    922         print('user-supplied args:', list(params.keys()))
--> 923 result = eval_func(self[variable], params)
    924 if isinstance(result, GeneratorType):
    925     if (len(params) == 1) & isinstance(list(params.values())[0], GeneratorType):

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/kamodo/util.py:272, in eval_func(func, kwargs)
    270 '''Evaluate function over valid argument'''
    271 try:
--> 272     return func(**valid_args(func, kwargs))
    273 except TypeError as m:
    274     raise TypeError(str(m) + str(list(valid_args(func, kwargs).keys())))

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/decorator.py:232, in decorate.<locals>.fun(*args, **kw)
    230 if not kwsyntax:
    231     args, kw = fix(args, kw, sig)
--> 232 return caller(func, *(extras + args), **kw)

File ~/.virtualenvs/kamdo/lib/python3.8/site-packages/kamodo/util.py:183, in decorator_wrapper(f, *args, **kwargs)
    181 def decorator_wrapper(f, *args, **kwargs):
    182     """Wrapper needed by decorator.decorate to pass through args, kwargs"""
--> 183     return f(*args, **kwargs)

Input In [6], in fvec(rvec)
      9 @kamodofy(units = 'km/s')
     10 def fvec(rvec = points):
---> 11     ux = np.sin(rvec[:,0])
     12     uy = np.cos(rvec[:,1])
     13     return np.vstack((ux,uy)).T

IndexError: too many indices for array: array is 0-dimensional, but 2 were indexed
```

**Comment by asher-pembroke** [comment]
The problem was this line

```python
points = np.array(zip(xx.ravel(), yy.ravel()))
```
In the python 2.7 days, this would have created a 2d array. Now you have to cast the zip into an array first

```python
points = np.array(list(zip(xx.ravel(), yy.ravel())))
```
After that, the rest of the code works. I'll update the docs to reflect this.

**Comment by asher-pembroke** [comment]
should be resolved by f6d149c
